ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -53,6 +52,33 @@ module Json = struct
   let merge_maps m maps =
     List.fold_left (fun acc map -> StringMap.union choose_enum acc map) m maps
 
+  let rec get_fp_type ty =
```

Comment:
Will there be other types like Hashtbl?

Reply:
No other types, all types of object here:
```ocaml
type ty =
  | SecretString
  | String
  | Int
  | Float
  | Bool
  | DateTime
  | Enum of string * (string * string) list
  | Set of ty
  | Map of ty * ty
  | Ref of string
  | Record of string
  | Option of ty
[@@deriving rpc]
```

Reply:
Thank you, I think those are only the types used in Xapi API?